middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -263,46 +263,46 @@ let arithmetic_conversion dbg src dst arg =
   let open K.Standard_int_or_float in
   match src, dst with
   (* Identity on floats *)
-  | Naked_float, Naked_float -> None, arg
+  | Naked_float, Naked_float -> None, Fun.id
   (* Conversions to and from tagged ints *)
   | ( (Naked_int32 | Naked_int64 | Naked_nativeint | Naked_immediate),
       Tagged_immediate ) ->
-    None, C.tag_int arg dbg
+    None, fun arg -> C.tag_int arg dbg
```

Comment:
This is going to create a lot of closures.  Is there any way of avoiding that?

Reply:
There indeed is, and I've just pushed commits that avoid the allocating such closures.